packages/mobile/src/send/utils.ts

```diff
@@ -184,28 +188,44 @@ export function* handleSendPaymentData(
 
   yield put(storeLatestInRecents(recipient))
 
-  if (data.amount && data.currencyCode) {
-    const currency = data.currencyCode as LocalCurrencyCode
+  if (data.amount) {
+    const currency = data.currencyCode
+      ? (data.currencyCode as LocalCurrencyCode)
+      : yield select(getLocalCurrencyCode)
     const exchangeRate: string = yield call(fetchExchangeRate, currency)
     const dollarAmount = convertLocalAmountToDollars(data.amount, exchangeRate)
     if (!dollarAmount) {
       Logger.warn(TAG, '@handleSendPaymentData null amount')
       return
     }
```

Comment:
I think this portion could be put inside the `else if (data.token === 'cUSD' || !data.token)` clause? Since these variables aren't used if `data.token === 'CELO'